swyft/utils/utils.py

```diff
@@ -145,8 +145,9 @@ def estimate_coverage(
     diags = []
     for _ in range(nrounds):
         for point in dataset:
-            samples = marginals(point["obs"], nsamples)
-            diag = sample_diagnostics(samples, true_params=point["par"])
+            samples = post.sample(nsamples, point[0])
+            diag = sample_diagnostics(samples, true_params=point[2])
```

Comment:
When I checked it, sample_diagnostics was not defined. It is currently commented out.

Also to be honest I find the function a bit confusing. Is it possible to break it up into each function you want it to preform then call that specific function, rather than having so much state control?

Reply:
`sample_diagnostics` is deprecated as well, or needs updating. I'll add a comment into the code.

Reply:
(that is why it was commented out BTW)